packages/react/passport/src/PassportContext.tsx

```diff
@@ -0,0 +1,187 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import React, {
+  createContext, useCallback, useContext, useEffect, useMemo,
+  useState,
+} from 'react';
+import { setPassportClientId, track } from '@imtbl/metrics';
+import {
+  Passport, PassportModuleConfiguration,
+  UserProfile,
+} from '@imtbl/passport';
+
+type ZkEvmReactContext = {
+  passportInstance: Passport;
+  isLoggedIn: boolean;
+  isLoading: boolean;
+  withWallet: boolean;
+  error: Error | null;
+  getIdToken: Passport['getIdToken'];
+  getAccessToken: Passport['getAccessToken'];
+  getAccounts: () => Promise<string[]>;
+  getLinkedAddresses: Passport['getLinkedAddresses'];
+  getProfile: () => Promise<UserProfile | undefined>;
+  login: (options?: {
+    useCachedSession?: boolean;
+    anonymousId?: string;
+    withoutWallet?: boolean;
+  }) => Promise<void>;
+  logout: () => Promise<void>;
+  loginCallback: () => Promise<void>;
+  linkExternalWallet: Passport['linkExternalWallet'];
+};
+
+const PassportContext = createContext<ZkEvmReactContext | null>(null);
+
+type ZkEvmReactProviderProps = {
+  config: PassportModuleConfiguration;
+  children: React.ReactNode;
+};
+
+export function ZkEvmReactProvider({ children, config }: ZkEvmReactProviderProps) {
+  setPassportClientId(config.clientId);
+  track('passport', 'ZkEvmProviderInitialised');
+  const [isLoggedIn, setLoggedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<Error | null>(null);
+  const [withWallet, setWithWallet] = useState(true);
+  const passportInstance = new Passport(config);
+
+  const checkLogggedIn = useCallback(async () => {
+    const p = await passportInstance.login({
+      useCachedSession: true,
+    });
+    setLoggedIn(!!p);
+  }, [passportInstance]);
```

Comment:
This has the potential to cause re-rendering issues for consumers. My assumption is that we expect marketplaces to consume the provider like this:

```
function MyComponent() {
  const { isLoading, isLoggedIn } = reactPassport.usePassport();
  
  if (isLoading) {
    return <LoadingAnimation />;
  }
  
  if (isLoggedIn) {
    return <SomeComponentForAuthenticatedUsers />;
  }

  return <PromptUserToLogin />;
}
```

In the above scenario, if `login` triggers a token refresh, `PromptUserToLogin` would render momentarily. Once the token refresh finishes and `login` returns true, `SomeComponentForAuthenticatedUsers` would then render.